tasks/pkgtask/pkgTask.go

```diff
@@ -19,61 +20,20 @@ import (
 type PkgActionType int
 
 const (
+	TaskTypePkgInstalled = "pkg.installed"
+	TaskTypePkgRemoved   = "pkg.removed"
+	TaskTypePkgUpgraded  = "pkg.uptodate"
+
 	ActionInstall PkgActionType = iota + 1
 	ActionUninstall
 	ActionUpdate
 )
 
-type PkgTaskBuilder struct {
-}
-
-var pkgTaskParamsFnMap = taskFieldsParserConfig{
-	NameField: {
-		parseFn: func(task Task, path string, val interface{}) error {
-			t := task.(*PkgTask)
-			t.Named.Name = fmt.Sprint(val)
-			return nil
-		},
-		fieldName: "Name",
-	},
-	NamesField: {
-		parseFn: func(task Task, path string, val interface{}) error {
-			var names []string
-			var err error
-			t := task.(*PkgTask)
-			names, err = conv.ConvertToValues(val)
-			t.Named.Names = names
-			return err
-		},
-		fieldName: "Names",
-	},
-}
-
-func (fmtb PkgTaskBuilder) Build(typeName, path string, params interface{}) (Task, error) {
-	task := &PkgTask{
-		TypeName: typeName,
-		Path:     path,
-	}
-
-	switch typeName {
-	case PkgInstalled:
-		task.ActionType = ActionInstall
-	case PkgRemoved:
-		task.ActionType = ActionUninstall
-	case PkgUpgraded:
-		task.ActionType = ActionUpdate
-	}
-
-	errs := Build(typeName, path, params, task, pkgTaskParamsFnMap)
-
-	return task, errs.ToError()
-}
-
-type PkgTask struct {
+type PTask struct {
```

Comment:
Any reason task structs are not just named `Task` but have the prefix? I would prefer either previous version with full name or just `Task`, this short version is not the clearest.

Reply:
not surprised that you didn't like these. but i have a big distrust of lots of types with the same name. i think they've caused to many headaches in the past. however, acknowledge that the golang way is to only use the package name for the name spacing, so the simple name is the way to go. the long form will attract the famous stutter warning. i've removed the short prefixes.